[ENH] Enhance `dt.fillna()` to support filling with a particular value (#3344)
h2oai/datatable: docs/api/dt/fillna.rst

    :src: src/core/expr/fexpr_fillna.cc pyfn_fillna
    :tests: tests/dt/test-fillna.py
    :cvar: doc_dt_fillna
    :signature: fillna(cols, value=None, reverse=False)

    .. x-version-added:: 1.1.0

    For each column from `cols` either fill the missing values with a single value, 

>     or fill with the previous or subsequent non-missing values. 

Reviewer: ```suggestion
    or fill with the previous/subsequent non-missing values. 
```

    In the presence of :func:`by()` the filling is performed group-wise.

    Parameters
    ----------
    cols: FExpr
        Input columns.

>     value: Scalar or FExpr to replace the missing values.

Reviewer: ```suggestion
    value: None | bool | int | float | str | list | tuple | dict | FExpr
        A single value, a list/tuple/dict of values or an f-expression (must be a reducer) to impute missing values with.
```

    :src: src/core/expr/fexpr_fillna.cc pyfn_fillna
    :tests: tests/dt/test-fillna.py
    :cvar: doc_dt_fillna
    :signature: fillna(cols, value=None, reverse=False)

    .. x-version-added:: 1.1.0

    For each column from `cols` either fill the missing values with a single value, 
    or fill with the previous/subsequent non-missing values. 
    In the presence of :func:`by()` the filling is performed group-wise.

    Parameters
    ----------
    cols: FExpr
        Input columns.

    value: None | bool | int | float | str | list | tuple | dict | FExpr

>         A single value, a list/tuple/dict of values or an f-expression (must be a reducer) to impute missing values with.

Reviewer: By looking at the implementation, it shouldn't be a reducer. Any f-expression will work, we just need to emphasize it in the docs and don't call it a "scalar".

```suggestion
        A single value, a list/tuple/dict of values or an f-expression to impute missing values with.
```
